README.md

```diff
@@ -11,135 +23,159 @@ Clone the depot_tools repository:
-
+#### 3.3 Build castanets for Ubuntu
+Build castanets (the “chrome” target) with Ninja using the command:
 ```sh
-$ gn gen out/Default
+$ ninja -C out/Default chrome
 ```
 
-You set build arguments on a build directory by typing:
 
+### (Optional) 4. Build Castanets for Android (Only a Renderer Process)
+For now, we need a **rooted Android phone** to run Android Renderers. 
+
+Install additional build dependencies
 ```sh
-$ gn args out/Default
+$ build/install-build-deps-android.sh
+```
+#### 4.1 Run the sync
+We need a gclient configuration. To create a .gclient file, run:
+```sh
+$ build/create_gclient.sh
 ```
 
-This will bring up an editor. Type build args into that file like this:
+#### 4.2 Add target OS(Android) Option
 
+We add target OS information to gclient.
 ```
-enable_castanets=true
-enable_nacl=false
-is_debug=false
+echo "target_os = [ 'android' ]" >> ../.gclient 
 ```
 
-For faster builds, the below can also be added in out/Default/args.gn
+Once you've run install-build-deps-android at least once, you can now run the Chromium-specific sync, which will download additional binaries and other things you might need:
+```sh
+$ gclient run-hooks
+$ gclient sync --with_branch_head
+```
+
+
+#### 4.3 Argument Setup for Build
 
+Chromium uses Ninja as its main build tool along with a tool called GN to generate .ninja files. You can create any number of build directories with different configurations. 
+To create a build directory, run:
+```sh
+$ gn gen --args='target_os="android" enable_castanets=true enable_nacl=false is_debug=false' out/Android
+```
+for faster build, you may add following options
 ```
 remove_webcore_debug_symbols=true
 is_component_build=true
 use_jumbo_build=true
 ```
 
-
-### Build castanets
-
-
+#### 4.4 Build castanets for Android
 Build castanets (the “chrome” target) with Ninja using the command:
-
 ```sh
-$ autoninja -C out/Default chrome
+$ ninja -C out/Android chrome_public_apk
 ```
 
 
-### Run castanets in a local machine (test only)
-
+### 5. How To run Castanets
+#### 5.1 Run castanets in a local Ubuntu machine 
 Start first chrome instance: Browser Process
-
 ```sh
 $ out/Default/chrome <URL>
 ```
 
-Start second chrome instance: Utility Process (Network Service)
-
-```sh
-$ out/Default/chrome --type=utility --server-address=127.0.0.1
-```
-
-Start third chrome instance: Renderer Process
-
+Start second chrome instance: Renderer Process
 ```sh
 $ out/Default/chrome --type=renderer --server-address=127.0.0.1
 ```
 
-### Run castanets in a distributed environment
-
-
-Device A: Browser Process
-
+#### 5.2 Run castanets in a distributed environment
+##### 5.2.1 Local Ubuntu Browser and Remote Ubuntu Renderer
+_Device A: Browser Process_
 ```sh
 $ out/Default/chrome <URL>
 ```
 
-Device B: Utility Process (Network Service)
+_Device B: Renderer Process_
+```sh
+$ out/Default/chrome --type=renderer --enable-castanets=<BROWSER IP ADDR>
```

Comment:
It doesn't support --enable-castanets in chromium_69.
--server-address=<BROWSER IP ADDR"

Reply:
Thank you. I missed that. Now modified.